app/controllers/api/v1/products_controller.rb

```diff
@@ -10,7 +11,7 @@ def index
   end
 
   def show
-    products = Product.find(params.require(:id))
+    products = @category.products.find(params.require(:id))
```

Comment:
I would instead add a before action only for `show`, `update`, and/or `destroy` methods that sets the `@product` instance.